### ISSUE TITLE
Two different scopes in template

### ISSUE THREAD

**Issue by alexr00** [jeff-hykin/cpp-textmate-grammar#274]
Sample C++
```cpp
    Task<ANY_OUTPUT_TYPE, ANY_INPUT_TYPE>::links_to;
```
The first `ANY_OUTPUT_TYPE` has the correct (or at least the same scope that it used to have):
```
storage.type.user-defined.cpp
meta.template.call.cpp
meta.template.call.cpp
source.cpp
source.cpp
```
The second on has a different scope:
```
entity.name.type.cpp
meta.qualified_type.cpp
meta.template.call.cpp
meta.template.call.cpp
source.cpp
source.cpp
```

**Comment by matter123** [comment]
@jeff-hykin Is the plan for template parameters in a templated type to be assumed that unless it is a literal the parameter is a type?

**Comment by jeff-hykin** [comment]
Yeah I think thats the best we can do. If variables are allowed, there's no way to know the difference and it would be extremely rare.

**Comment by matter123** [comment]
It would be a bit of a hack but if the parameter is not in the form `qualified_type.then(std_space).then(/[,>]/)` It is treated as an expression.

That would lead to multi-line type expressions being treated as regular expressions, however.

**Comment by jeff-hykin** [comment]
mutli-line template calls are already an issue so even that wont really be a downgrade.

**Comment by jeff-hykin** [comment]
fixed with `v1.12.12` 

also just as a note the storage.type.user-defined scope was something I created early on before finding out that it would be more consistent to tag it as `entity.name.type`. This was the last usage of the `storage.type.user-defined` so it shouldn't be a problem moving forward though.

**Comment by alexr00** [comment]
Pulled into VS Code insiders. Thank you!
